### ISSUE TITLE
strange temporary display issues

### ISSUE THREAD

**Issue by mikshaw** [mypaint/mypaint#282]
Since the last two updates (i do a git pull just about every day) i've noticed a bizarre thing happening when i first start mypaint. It happens every time.

* First thing i notice is that the canvas flashes white and then turns gray (possibly the default gtk theme background?).
* Next i can see that what i draw on the canvas is not being updated in the preview window.
* If i press F12, the lines that were made in the center of the canvas get shifted to the bottom right corner, and what i draw afterward now will show in the preview, although it bleeds beyond the edges of the preview window. The attached screenshot displays this.
* If i start a new document everything seems to work properly.
* If i save the current weird document and then open it, everything looks and behaves fine.

Basically it comes down to nothing serious, and easily dealt with, but it's just weird.

Debian Testing, 64-bit, Gtk 3.14.5, dwm window manager, Xorg 1.16.4, Intel HD4000 gpu

![15-0401-200735-window](https://cloud.githubusercontent.com/assets/5081349/6955662/5468d73e-d8af-11e4-8388-9c8961b98523.jpg)

**Comment by achadwick** [comment]
Cannot reproduce here. I need better steps to reproduce, ideally from a fresh config. But that can wait...

First, can you reliably tie it down to a specific recent commit? Try using a backup config with each one in turn back ten revisions or so.

```
$ cp ~/.config/mypaint/settings.json /tmp/settings.json.BAK    # NOTE: may still be ~/.mypaint
$ cd path/to/mypaint
$ scons --clean
$ git checkout master
$ git submodule update --force
$ scons
$ ./mypaint
```

Is it still happening after this clean build? If so, for each recent revision

```
$ scons --clean
$ git checkout 1234567890abcdef1234567890abcdef    # revision checksum goes here!
$ git submodule update --force
$ scons
$ cp /tmp/settings.json.BAK ~/.config/mypaint/settings.json       # targ may may to be be ~/.mypaint
$ ./mypaint
```

and let me know which revision introduces the problem. Due to the submodule bump, you'll need the submodule update on each round just to be sure.

If you know how to use `git bisect`, then please use that and paste the results. But please use the same config on each testing round if you can.

**Comment by achadwick** [comment]
I should add: I can't reproduce this in GNOME, with most other things being approximately the same. Can you try under GNOME or Xfce to exclude dwm as the source of it please?

**Comment by mikshaw** [comment]
I was still having trouble after a clean build.
I didn't think to try a clean config, and it works now...apparently the trouble is in there somewhere.
Thank you.

**Comment by mikshaw** [comment]
I deleted the scratchpads and now everything seems to be working again.

**Comment by achadwick** [comment]
It might be nice to know what sort of broken scratchpad file was causing the issue. It'd be the scratchpads/autosave.ora one if you still have a copy.

**Comment by mikshaw** [comment]
I failed to report that i also deleted the background directory, which contained only the default background tile. I could see a corrupted background tile being a much more likely culprit than the scratchpads in this case. Sorry that slipped my mind...i guess i was excited that it worked so much more quickly than expected.

I just started deleting the least important stuff first, with the intention of removing everything if necessary, with little concern about backing it up. Backing up app configurations was never very important to me unless it's a complicated thing to do manually, so it didn't even occur to me to save a copy. I can tell you that there were three files in the scratchpads directory, and one of them was a .BAK

I get this feeling that i did something wrong just before the problem occurred, but i don't remember what. It seems like the program worked properly for one day in between a git pull and when the problem started, but i'm not sure of that either.  I vaguely remember feeling an "oops" moment at one time and then ignoring it, but that could be confirmation bias.

One unusual thing i did at that time was run a build test, which i assume is unrelated, but since i'm not much of a programmer i don't know. I made a copy of the source, changed a file (attempting to force a build error), and ran scons. After seeing the error i deleted that copy. I doubt it has anything to do with it, but since it happened around the same time i figured i'd mention it.
